### ISSUE TITLE
Overlay map cannot be disabled

### ISSUE THREAD

**Issue by DmitryAlexei** [osmandapp/OsmAnd#18525]
### Description

One of OsmAnd users encountered a problem: he cannot disable overlay map.

### Steps to reproduce

1. Enable overlay map
2. Try to disable overlay map using one of the available options.

### Actual result

User claims, that he cannot disable overlay map using either the switch in the configure map menu or in the Overlay map. 

<table style="width:100%">
  <tr>
<th><img src="https://github.com/osmandapp/OsmAnd/assets/116079541/79449395-040d-4db6-9cf8-6de4585d82f4"  width="300" height="auto"/> </th>
 <th><img src="https://github.com/osmandapp/OsmAnd/assets/116079541/152cff50-3705-46ff-aae3-4add936d6035"  width="300" height="auto"/> </td>
  </tr>
</table>

### Expected result

Overlay map can be disabled

### Your Environment (required)

**WARNING** Crash-Logs **MAY** contain information you deem sensitive.
Review this **CAREFULLY** before posting your issue!

```
OsmAnd Version:
Android/iOS version:
Device model:
Crash-Logs: ?
```

**Comment by Chumva** [comment]
It looks like this behavior happens if we have map == null [here](https://github.com/osmandapp/OsmAnd/blob/0a93a61683f28bc34d7cc0f62080d13eb786bb48/OsmAnd/src/net/osmand/plus/plugins/rastermaps/OsmandRasterMapsPlugin.java#L632), but we need to investigate why it happens

**Comment by zipav** [comment]
OsmAnd Nightly 4.7.0#1272m, released: 2023-12-25

Difficult to reproduce. I tried different options, with different maps somehow:
Just switch the slider immediately, it successfully switches off. 
Close the app and open it again.
Change the transparency of the overlay before trying to click the slider.
Tried Microsoft Earth and Yandex RU maps.
Swiped the app to the background and after 5 minutes tried opening it and clicking the slider. In each of these cases, the switch successfully disabled the Overlay option.
